server/src/main/java/org/elasticsearch/index/mapper/NestedObjectMapper.java

```diff
@@ -441,7 +441,7 @@ public DocValuesLoader docValuesLoader(LeafReader leafReader, int[] docIdsInLeaf
 
         private List<Integer> collectChildren(int parentDoc, BitSet parentDocs, DocIdSetIterator childIt) throws IOException {
             assert parentDocs.get(parentDoc) : "wrong context, doc " + parentDoc + " is not a parent of " + nestedTypePath;
-            final int prevParentDoc = parentDocs.prevSetBit(parentDoc - 1);
+            final int prevParentDoc = parentDoc > 0 ? parentDocs.prevSetBit(parentDoc - 1) : -1;
```

Comment:
This is a separate bug, stole the fix from #110524. Needs to be extracted into a separate PR. We hit it a lot in tests now.

Reply:
https://github.com/elastic/elasticsearch/pull/112005

Reply:
Wanna add a test for it? I can take on it in #110524, otherwise..

Reply:
#112043